First view to Top Movies app

olegdaniv opened the pull request:
I have written this app to learn and remind  some features like:
- сustom view
- recycleView
- viewModel

Commit:
firs view to app

MonstraL commented on app/src/main/AndroidManifest.xml:
```diff
@@ -2,7 +2,7 @@
     xmlns:tools="http://schemas.android.com/tools"
     package="com.example.topmovies">
-
+<uses-permission android:name="android.permission.INTERNET"/>
```
Please format the code. There must be the tab space here

olegdaniv replied:
ok, I forgot to do com + alt + L to my manifest file

MonstraL commented on app/src/main/java/com/example/topmovies/adapter/MovieAdapter.kt:
```diff
@@ -0,0 +1,64 @@
+import com.example.topmovies.unit.REPLACE_AFTER
+
+class MovieAdapter : RecyclerView.Adapter<MovieAdapter.MainViewHolder>() {
+    var movies = mutableListOf<Movie>()
```
It is better to replace with an emptyList

olegdaniv replied:
OK, but I don't understand why it's so important?

MonstraL replied:
I see no reason to make a mutableList here. It's better not to give the ability to change lists where it's not necessary

MonstraL commented on app/src/main/java/com/example/topmovies/adapter/MovieAdapter.kt:
```diff
@@ -0,0 +1,64 @@
+class MovieAdapter : RecyclerView.Adapter<MovieAdapter.MainViewHolder>() {
+    var movies = mutableListOf<Movie>()
+    fun setMovieList(movies: List<Movie>) {
+        this.movies = movies.toMutableList()
```
Why do you need a MutableList here?

olegdaniv replied:
Sure, in this case, I don't need MutableList.

MonstraL commented:
https://kotlinlang.org/docs/coding-conventions.html
@OlegDaniv Please read these rules and edit your code

MonstraL commented on app/src/main/java/com/example/topmovies/view/CircleAvatarView.kt:
```diff
@@ -0,0 +1,107 @@
+        setLayerType(LAYER_TYPE_SOFTWARE, null)
+
+        if (attributeSet != null) {
+            val ta =
```
"The names should make it clear what the purpose of the entity is"

olegdaniv replied:
I'll consider that.

Commit:
- clean code
- replaced the movies mutableListOf field to ListOf
- rename field ta to typedArray

MonstraL commented on app/src/main/java/com/example/topmovies/retrofit/ApiServer.kt:
```diff
@@ -0,0 +1,29 @@
+    fun getAllMovies(): Call<MovieObject>
+
+    companion object {
+        var ApiService: ApiServer? = null
```
This is a violation of naming conventions

olegdaniv replied:
Alright, what do you think is a good name for this class?

MonstraL replied:
The problem is not the class name. Variable name must start with a lowercase

olegdaniv replied:
Yeah, sorry, I missed that one

Commit:
- clean code

Commit:
- optimized circleAvatarView
- chanted def image
- clean code

Commit:
- remove Paint.Align.CENTER, because it default

Commit:
- fixed recyclerview blinking
- added cardView
- clean code

Commit:
- added SwipeRefreshLayout
- created field avatarCustomTarget
- clean code

YuriiTsap commented on app/src/main/java/com/example/topmovies/unit/Constants.kt:
```diff
@@ -0,0 +1,6 @@
+const val BASE_URL = "https://imdb-api.com/"
+const val TOP_100_MOVIES = "en/API/MostPopularMovies/k_wfrg16mx"
+const val REPLACE_AFTER = "._V1_"
+const val IMAGE_SIZE = "UX126_CR6,0,126,126_AL_.jpg"
```
new line

YuriiTsap commented on app/src/main/res/layout/item_layout.xml:
```diff
@@ -0,0 +1,83 @@
+    android:layout_marginEnd="@dimen/layout_margin"
+    app:cardBackgroundColor="@color/mint_macaron"
+    app:cardCornerRadius="@dimen/item_cardView_corner_radius"
+    app:cardElevation="5dp"
+    app:cardMaxElevation="5dp"
+    app:contentPadding="10dp"
+    app:contentPaddingBottom="10dp">
```
- extract to values
- should be dividable by 2

Commit:
- optimized code

Commit:
- optimized code
- remove default image in avatarView

Commit:
- added movie detailed Activity
